### PR TITLE
V2.1.3

### PR DESCRIPTION
**Pull request by sasbury**
* [Added] Methods to include error and connection listener with factory #113
* [Fixed] Protocol buffer build issues #112
* [Fixed] Made use of ossh password optional #111
* [Fixed] Typo in readme pointing to java repo #110

### CODE REVIEW

File: src/main/java/io/nats/streaming/StreamingConnectionFactory.java [hunk 1]
```diff
@@ -242,4 +247,38 @@ public void setClusterId(String clusterId) {
...
+    /**
+     * @return the connection listener configured for this factory
+     */
+    public ConnectionListener getConnectionListener() {
+        return this.connListener;
+    }
+
+    /**
+     * Set a connection listener for the underlying nats connection.
+     *
```
Comment (davidkemper) [inline src/main/java/io/nats/streaming/StreamingConnectionFactory.java]: extra blank comment line.

File: build.gradle [hunk 2]
```diff
@@ -191,7 +194,7 @@ uploadArchives {
             repository(url: "file://$buildDir/repos")
             if (project.hasProperty("local_archives") || ("true".equals(System.getenv("TRAVIS_PULL_REQUEST")))) {
                 repository(url: "file://$buildDir/repos")
-            } else {
+            } else if (project.hasProperty('ossrhUsername') && project.hasProperty('ossrhPassword')) {
```
Comment (davidkemper) [inline build.gradle]: The code above uses `"` for strings, and here you are using `'`.

More globally, should these magic names be named/documented constants somewhere?

File: src/main/java/io/nats/streaming/StreamingConnectionFactory.java [hunk 3]
```diff
@@ -58,7 +63,7 @@ public StreamingConnection createConnection() throws IOException, InterruptedExc
     Options options() {
         return new Options.Builder().connectWait(connectTimeout).pubAckWait(ackTimeout)
                 .discoverPrefix(discoverPrefix).maxPubAcksInFlight(maxPubAcksInFlight)
-                .natsConn(natsConn).natsUrl(natsUrl).build();
+                .natsConn(natsConn).natsUrl(natsUrl).connectionListener(connListener).errorListener(errListener).build();
```
Comment (davidkemper) [inline src/main/java/io/nats/streaming/StreamingConnectionFactory.java]: Do these new init stubs require a test as well?

Comment (sasbury) [reply src/main/java/io/nats/streaming/StreamingConnectionFactory.java]: should be tested in the options tests

File: src/main/java/io/nats/streaming/StreamingConnectionFactory.java (hunk 1 above)
Comment (sasbury) [reply src/main/java/io/nats/streaming/StreamingConnectionFactory.java]: i like those, but removed for you

File: build.gradle (hunk 2 above)
Comment (sasbury) [reply build.gradle]: it appears to be a mishmash, i am hesitant to fix/refactor until i can do a full clean up.

**Pull request merged**